neo/Prompt/Commands/BuildNRun.py

```diff
@@ -67,53 +86,48 @@ def BuildAndRun(arguments, wallet, verbose=True, min_fee=DEFAULT_MIN_FEE, invoca
-            except Exception as e:
-                logger.error('Could not convert result to ContractParameter: %s ' % e)
-
-            if tx is not None and result is not None:
-                if verbose:
-                    print("\n-----------------------------------------------------------")
-                    print("Calling %s with arguments %s " % (path, [item for item in reversed(engine.invocation_args)]))
-                    print("Test deploy invoke successful")
-                    print("Used total of %s operations " % total_ops)
-                    print("Result %s " % return_type_results)
-                    print("Invoke TX gas cost: %s " % (tx.Gas.value / Fixed8.D))
-                    print("-------------------------------------------------------------\n")
-
-                return tx, result, total_ops, engine
-            else:
-                if verbose:
-                    print("Test invoke failed")
-                    print("tx is, results are %s %s " % (tx, result))
 
+    if wallet is not None:
+
+        f_args = arguments[1:]
+        i_args = arguments[6:]
+
+        script = GatherLoadedContractParams(f_args, contract_script)
+
+        tx, result, total_ops, engine = test_deploy_and_invoke(script, i_args, wallet, from_addr,
+                                                               min_fee, invocation_test_mode, debug_map=debug_map,
+                                                               invoke_attrs=invoke_attrs, owners=owners)
+        i_args.reverse()
+
+        return_type_results = []
+        try:
+            rtype = ContractParameterType.FromString(f_args[1])
+            for r in result:
+                cp = ContractParameter.AsParameterType(rtype, r)
+                return_type_results.append(cp.ToJson())
+        except Exception as e:
+            print('Could not convert result to ContractParameter: %s ' % e)
+
+        if tx is not None and result is not None:
+            if verbose:
+                print("\n-----------------------------------------------------------")
+                print("Calling %s with arguments %s " % (path, [item for item in reversed(engine.invocation_args)]))
+                print("Test deploy invoke successful")
+                print("Used total of %s operations " % total_ops)
+                print("Result %s " % return_type_results)
+                print("Invoke TX gas cost: %s " % (tx.Gas.value / Fixed8.D))
+                print("-------------------------------------------------------------\n")
+
+            return tx, result, total_ops, engine
         else:
-
-            print("please open a wallet to test built contract")
-
-    return None, None, None, None
+            if verbose:
+                print("Test invoke failed")
+                print("tx is, results are %s %s " % (tx, result))
+            return tx, result, None, None
+
+    else:
+        print("Please open a wallet to test build contract")
```

Comment:
We can remove a level of indentation if we change (line 90)
```
     if wallet is not None:
 
          f_args = arguments[1:]
```
to something like
```
if not wallet:
    return None, None, None, None

f_args = arguments[1:]
etc.
```